eng/common/scripts/Set-VcpkgWriteModeCache.ps1

```diff
@@ -0,0 +1,46 @@
+#!/bin/env pwsh
+param(
+  [string] $StorageAccountName = 'azuresdkartifacts',
+  [string] $StorageContainerName = 'public-vcpkg-container'
+)
+
+. "$PSScriptRoot/Helpers/PSModule-Helpers.ps1"
+
+Write-Host "`$env:PSModulePath = $($env:PSModulePath)"
+
+# Work around double backslash
+if ($IsWindows) {
+  $hostedAgentModulePath = $env:SystemDrive + "\\Modules"
+  $moduleSeperator = ";"
+}
+else {
+  $hostedAgentModulePath = "/usr/share"
+  $moduleSeperator = ":"
+}
+$modulePaths = $env:PSModulePath -split $moduleSeperator
+$modulePaths = $modulePaths.Where({ !$_.StartsWith($hostedAgentModulePath) })
+$AzModuleCachePath = (Get-ChildItem "$hostedAgentModulePath/az_*" -Attributes Directory) -join $moduleSeperator
+if ($AzModuleCachePath -and $env:PSModulePath -notcontains $AzModuleCachePath) {
+  $modulePaths += $AzModuleCachePath
+}
```

Comment:
Why is all this needed? It also looks like we have similar logic in PSModule-Helpers.ps1